### DISCUSSION TITLE
Shopping List UX improvement

### DISCUSSION THREAD

**Discussion by abaisero** [mealie-recipes/mealie#2701] [Feature Request]
### First Check

- [X] I used the GitHub search to find a similar requests and didn't find it.
- [X] Checked the [tasks tagged](https://github.com/hay-kot/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered

### Please provide a concise description of the problem that would be addressed by this feature.

The current layout of the shopping lists shows the checkbox and ingredient name on the left, and then the click-n-drag and edit icons far away on the right, separated by a fairly wide margin.  This makes it quite hard hard to make sure that you're dragging or editing the correct item.

Especially since the ingredients in a shopping list are not automatically ordered by type or isle, reordering them beforehand is very helpful.

### Please provide a concise description of the feature that would resolve your issue.

A very simple change to fix this issue would be to move the drag and edit icons to the left, such that they stand right next to their respective ingredients, making a lot simpler to select the correct one.

![image](https://github.com/mealie-recipes/mealie/assets/1794938/d9487c5f-f89e-422e-be72-a2dfa5a1b4c6)

### Please consider and list out some caveats or tradeoffs made in your design decision

Maybe there could be some concern that this makes it more likely to click the checkbox by accident;  but I think this could be minimized by adding only a little bit of space between the controls, rather than the huge space that now exists between some controls and the respective text.

### Additional Information

- [ ] If this is accepted I'm willing to submit a PR to provide this feature
- [ ] If this is accepted I'm willing to help maintain this feature
- [ ] I'm willing to sponsor/pay a developer to do this work

**Comment by michael-genson** [comment]
> the ingredients in a shopping list are not automatically ordered by type or isle

FYI you can group shopping list items by label, and you can sort your labels so they match the layout of your supermarket/grocery store:
![image](https://github.com/mealie-recipes/mealie/assets/71845777/8999b8f8-8b7f-44ec-a0f6-28c0d4ce65d9)

![image](https://github.com/mealie-recipes/mealie/assets/71845777/075eb5d8-66ea-479c-ac39-e7a7ebecacba)

I think this FR is still valid, just wanted to make you aware of the sorting features!

**Reply by abaisero** [reply]
Oh!  I did not realize there was such a functionality;  I'll check it out and revaluate my suggestion, thanks!

**Reply by abaisero** [reply]
OK, yes, that is definitely useful for my purposes;  but yeah as you also say, I think the FR is still valid.
